controllers/config_map.go

```diff
@@ -47,70 +49,149 @@ type ConfigMapReconciler struct {
 	AssumeRole   string
 }
 
+type ClusterInfo struct {
+	Name           string
+	Namespace      string
+	Region         string
+	AWSPartition   string
+	VpcID          string
+	RoleArn        arn.ARN
+	SecurityGroups *crossplaneConfigValuesAWSClusterSecurityGroups
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&capa.AWSCluster{}).
+		For(&capi.Cluster{}).
 		Complete(r)
 }
 
 func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
 
-	cluster := &capa.AWSCluster{}
-	err := r.Client.Get(ctx, req.NamespacedName, cluster)
+	awsCluster := &capa.AWSCluster{}
+	ekscontrolplane := &eks.AWSManagedControlPlane{}
```

Comment:
Please put those where they're used (`if` block) so they can't be used uninitialized in a too-wide scope.